File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.cs
using System;
using System.Linq;
using Mono.Data.Sqlite;

public class Sample
{
    string ConstQuery = "";

    void Compliant(SqliteConnection connection)
    {
        var command = new SqliteCommand();       // Compliant
        command = new SqliteCommand(connection); // Compliant
        var adapter = new SqliteDataAdapter();   // Compliant
    }

    void Foo(SqliteConnection connection, string query, SqliteTransaction transaction, params object[] parameters)
    {
        var command = new SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}", connection);          // Noncompliant
        command = new SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}");                          // Noncompliant
        command = new SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}", connection, transaction); // Noncompliant
        var adapter = new SqliteDataAdapter(string.Concat(query, parameters), connection);                  // Noncompliant
        adapter = new SqliteDataAdapter(string.Concat(query, parameters), "connection");                    // Noncompliant
    }

    // https://github.com/SonarSource/sonar-dotnet/issues/7261
    void Reproduce_7261(string connectionString, string query)
    {
        string sql = "select * from table where query = '" + query + "';"; // Secondary [adapter, command]

        using (SqliteConnection connection = new SqliteConnection(connectionString))
        {
            connection.Open();

            var adapter = new SqliteDataAdapter(sql, connection); // Noncompliant [adapter]
            var command = new SqliteCommand(sql, connection);     // Noncompliant [command]
        }
    }
}

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.vb
Imports System
Imports System.Linq
Imports Mono.Data.Sqlite

Namespace Test
    Public Class C
        Private ConstQuery As String = ""

        Private Sub Compliant(ByVal connection As SqliteConnection)
            Dim command = New SqliteCommand()
            command = New SqliteCommand(connection)
            Dim adapter = New SqliteDataAdapter()
        End Sub

        Private Sub Foo(ByVal connection As SqliteConnection, ByVal query As String, ParamArray parameters As Object())
            Dim command = New SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}", connection)
            command = New SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}")
            Dim adapter = New SqliteDataAdapter(String.Concat(query, parameters), connection)
        End Sub
    End Class
End Namespace
